pkg/module/sync/runner.go

```diff
@@ -110,23 +110,37 @@ func (r *Runner) updateManifests(ctx context.Context, kyma *v1beta2.Kyma,
 		return commonerrs.ErrTypeAssert
 	}
 
-	if err := r.doUpdateWithStrategy(ctx, kyma.Labels[shared.ManagedBy], module.Enabled,
-		manifestObj); err != nil {
+	if err := r.DoUpdateWithStrategy(ctx, kyma.Labels[shared.ManagedBy], module.Enabled, manifestObj); err != nil {
 		return err
 	}
 	module.Manifest = manifestObj
 	return nil
 }
 
-func (r *Runner) doUpdateWithStrategy(ctx context.Context, owner string, isEnabledModule bool,
+func (r *Runner) DoUpdateWithStrategy(ctx context.Context, owner string, isEnabledModule bool,
 	manifestObj *v1beta2.Manifest,
 ) error {
+	manifestInCluster := &v1beta2.Manifest{}
+	var err error
+	if err = r.Get(ctx, client.ObjectKey{Namespace: manifestObj.GetNamespace(), Name: manifestObj.GetName()},
```

Comment:
we should not do `Get` here, which means that we introduce an additional request API server for each `manifestObj`, which means around 10k requests more for the current PROD landscape, please think twice when introducing new requests to the API server, it will affect performance hugely.